bin/systemml-spark-submit.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python
+#-------------------------------------------------------------
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+#-------------------------------------------------------------
+
+import os
+import sys
+from os.path import join, exists
+from os import environ
+import argparse
+
+
+# error help print
+def print_usage_and_exit():
+    this_script = sys.argv[0]
+    print('Usage: ' + this_script + ' <dml-filename> [arguments]')
+    sys.exit(1)
+
+if environ.get('SPARK_HOME') is None:
+    print('SPARK_HOME not set')
+
+if len(sys.argv) < 2:
+    print('Wrong usage')
+    print_usage_and_exit()
+
+parser = argparse.ArgumentParser(description='System-ML Spark Submit Script', add_help=False, formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+parser.add_argument('--help', action='help', help='Print this usage message and exit')
+parser.add_argument('--master', default='local[*]', help='local, yarn-client, yarn-cluster', metavar='\b')
+parser.add_argument('--driver-memory', default='5G', help='Memory for driver (e.g. 512M)', metavar='\b')
+parser.add_argument('--num-executors', default='2', help='Number of executors to launch', metavar='\b')
+parser.add_argument('--executor-memory', default='2G', help='Memory per executor', metavar='\b')
+parser.add_argument('--executor-cores', default='1', help='Number of cores', metavar='\b')
+parser.add_argument('--conf', default='', help='Configuration settings', metavar='\b')
+parser.add_argument('-f', required=True, help='DML script file name', metavar='\b')
```

Comment:
There are 2 distinct set of arguments - one set to spark (master, executors...) and one set to systemml (nvargs, args, explain, stats...). In the python script, separate them out visually, add a newline with relevant comments.